src/routes/upload/index.tsx

```diff
@@ -0,0 +1,254 @@
+import React, { ReactElement } from 'react'
+import { Grid, Input, Button, Select } from '@material-ui/core'
+import NavBar from '../../components/NavBar'
+import { ENDPOINT } from '../../utils/config'
+import FileInput from '../../components/FileInput'
+import './main.sass'
+
+function isLIURL(str: string) {
+  const pattern = new RegExp(
+    '^(https?:\\/\\/)?' + // protocol
+    '(((linkedin)|(www)).)+com/in' + // domain name
+      '(\\/[-a-z\\d%_.~+]+)$', // port and path
+    'i'
+  ) // fragment locator
```

Comment:
```suggestion
  const pattern = new RegExp(/^((https:\/\/)?(www)?(linkedin)\.com\/in\/[-a-z\d%_.~+]+)$/); 
```
Does this look equivalent/okay to you? I think it's way more readable this way, but let me know if I messed up in simplifying/changing it.
This should allow urls that are:
linkedin.com/in/asdasd
www.linkedin.com/in/asdasd
https://linkedin.com/in/asdasdasd
https://www.linkedin.com/in/asdasds

Reply:
smh you forgot a \.

Reply:
nvm regex doesnt work without a loader and i'm lazy but i used your pattern cause it's better